fe/fe-core/src/main/java/org/apache/doris/nereids/analyzer/UnboundRelation.java

```diff
@@ -44,6 +45,7 @@
  */
 public class UnboundRelation extends LogicalRelation implements Unbound {
 
+    public final TableSample tableSample;
```

Comment:
use Optional to avoid use null directly

Reply:
According to the one who proposed Optional api to Java8, seems we should almost never wrap a member with Optional, the only scenario use this API is the return type of a function https://stackoverflow.com/questions/26327957/should-java-8-getters-return-optional-type/26328555#26328555

Reply:
I checked the codes by executing command `grep -r 'private Optional' .` under org/apache/doris/nereids package, there's only few fields declared as `Optional<SomeFields> aFields;`

Reply:
if u insist on using null, i cannot merge this PR. because NPE is hard to follow and the error message is useless for debuging